ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/desc/DescTableOperation.java

```diff
@@ -217,9 +216,7 @@ private void getColumnDataColPathSpecified(Table table, Partition part, List<Fie
       }
     } else {
       List<String> partitions = new ArrayList<String>();
-      // The partition name is converted to lowercase before generating the stats. So we should use the same
-      // lower case name to get the stats.
-      String partName = HMSHandler.lowerCaseConvertPartName(part.getName());
+      String partName = part.getName();
```

Comment:
`part` is returned from HMS, so the partition key is always lower case and do not need this convert.

Reply:
I'm not sure I understand the change here. But Hive supports the case sensitive on partition names.
create table testpartbl(id int) partitioned by(dt string);
```
alter table testpartbl add partition(dt='AA');
alter table testpartbl add partition(dt='aa');
alter table testpartbl add partition(dt='Aa');
```
 `show partitions testpartbl;`
```
+------------+
| partition  |
+------------+
| dt=AA      |
| dt=Aa      |
| dt=aa      |
+------------+
```

Reply:
Yes, partition value is case sensitive, `HMSHandler#lowerCaseConvertPartName` will only convert partition keys.

Reply:
i would keep it unchanged here, as it's outside of the current PR scope + `lowerCaseConvertPartName` is used in multiple similar places like `HiveMetaStore#get_partitions_statistics`

Reply:
disregard the above comment, looks like you have already removed `lowerCasePartNames` in that PR

Reply:
The `lowerCaseConvertPartName` is only used in partition stats related APIs. I think it's OK to remove it, because most of APIs such as `HMSHandler#get_partitions_by_names` related to partition names do not contain such behavior.

In other words, HMS requires that user should input the correct partition names to get the result.

Reply:
I would prefer leaving the `lowerCaseConvertPartName` as it is in `HMSHandler` if it not causes any big issues, as there are some other engines relying on Hive to get the stats, but it could be removed from `DescTableOperation`